Stratis.Bitcoin/Connection/ConnectionManager.cs

```diff
@@ -236,50 +236,68 @@ private string GetVersion()
 
 		public void Dispose()
 		{
-			if (DiscoveredNodeGroup != null)
-				DiscoveredNodeGroup.Dispose();
-			if (ConnectNodeGroup != null)
-				ConnectNodeGroup.Dispose();
-			if (AddNodeNodeGroup != null)
-				AddNodeNodeGroup.Dispose();
-			foreach (var server in _Servers)
+			if (this.DiscoveredNodeGroup != null)
+				this.DiscoveredNodeGroup.Dispose();
+			if (this.ConnectNodeGroup != null)
+				this.ConnectNodeGroup.Dispose();
+			if (this.AddNodeNodeGroup != null)
+				this.AddNodeNodeGroup.Dispose();
+			foreach (NodeServer server in this._Servers)
 				server.Dispose();
-			foreach (var node in ConnectedNodes.Where(n => n.Behaviors.Find<ConnectionManagerBehavior>().OneTry))
+			foreach (Node node in this.ConnectedNodes.Where(n => n.Behaviors.Find<ConnectionManagerBehavior>().OneTry))
 				node.Disconnect();
 		}
 
-
 		private readonly NodesCollection _ConnectedNodes = new NodesCollection();
-		public NodesCollection ConnectedNodes
+		public IReadOnlyNodesCollection ConnectedNodes { get { return this._ConnectedNodes; } }
+
+		internal void AddConnectedNode(Node node)
 		{
-			get
-			{
-				return _ConnectedNodes;
-			}
+			this._ConnectedNodes.Add(node);
+		}
+
+		internal void RemoveConnectedNode(Node node)
+		{
+			this._ConnectedNodes.Remove(node);
+		}
+
+		public Node FindNodeByEndpoint(IPEndPoint endpoint)
+		{
+			return this._ConnectedNodes.FindByEndpoint(endpoint);
+		}
+
+		public Node FindNodeByIp(IPAddress ip)
+		{
+			return this._ConnectedNodes.FindByIp(ip);
+		}
+
+		public Node FindLocalNode()
+		{
+			return this._ConnectedNodes.FindLocal();
 		}
 
-		public void AddNode(IPEndPoint endpoint)
+		public void AddNodeAddress(IPEndPoint endpoint)
 		{
-			var addrman = AddressManagerBehavior.GetAddrman(AddNodeNodeGroup.NodeConnectionParameters);
+			AddressManager addrman = AddressManagerBehavior.GetAddrman(this.AddNodeNodeGroup.NodeConnectionParameters);
 			addrman.Add(new NetworkAddress(endpoint));
-			AddNodeNodeGroup.MaximumNodeConnection++;
+			this.AddNodeNodeGroup.MaximumNodeConnection++;
 		}
 
-		public void RemoveNode(IPEndPoint endpoint)
+		public void RemoveNodeAddress(IPEndPoint endpoint)
 		{
-			Node node = this.ConnectedNodes.FindByEndpoint(endpoint);
+			Node node = this._ConnectedNodes.FindByEndpoint(endpoint);
 			if (node != null)
 				node.DisconnectAsync("Requested by user");
 		}
 
 		public Node Connect(IPEndPoint endpoint)
 		{
-			var cloneParameters = _Parameters.Clone();
+			NodeConnectionParameters cloneParameters = this._Parameters.Clone();
```

Comment:
there are mixed conventions for private field backers throughout
I think:
		NodeConnectionParameters _Parameters;
should be:
		NodeConnectionParameters parameters;
According to style guide. 

Didn't realize. I'll fix these.